File: Chase/AlmostEquivalentStrings.js
function areAlmostEquivalent (s,t) {
    const finalArr = [];

    for (let i = 0; i < s.length; i++) {
        const sCurrentArray = s[i].split("");
        const tCurrentArray = t[i].split("");
        const sLength = sCurrentArray.length;
        const tLength = tCurrentArray.length;
        const sMap = new Map();
        const tMap = new Map();

        if (sLength !== tLength) {
            finalArr.push('NO');
            break;
        }

        for (let j = 0; j < sCurrentArray.length; j++) {
            const sElement = sCurrentArray[j];
            const tElement = tCurrentArray[j];
            
            if (!sMap.has(sElement)) {
                sMap.set(sElement, 1);
            } else {
                sMap.set(sElement, sMap.get(sElement) + 1);
            }

            if (!tMap.has(tElement)) {
                tMap.set(tElement, 1);
            } else {
                tMap.set(tElement, tMap.get(tElement) + 1);
            }
        }

        for (const [element, value] of sMap) {
            let tValue = tMap.get(element);

            if (!tMap.has(element)) {
                tValue = 0;
            }

            const difference = Math.abs(value - tValue);

            if (difference <= 3) {
                continue;
            } else if (difference > 3) {
                finalArr.push('NO');
                break;
            }
            finalArr.push('YES');
        }
        
        sMap.clear();
        tMap.clear();
    }
    return finalArr
}

console.log(areAlmostEquivalent(['aabaab', 'aaaaabb'], ['bbabbc', 'abb']))

File: Chase/AlmostEquivalentStrings.py
def areAlmostEquivalent(s, t):
    final_arr = []

    for i in range(len(s)):
        s_current_array = list(s[i])
        t_current_array = list(t[i])
        s_length = len(s_current_array)
        t_length = len(t_current_array)
        s_map = {}
        t_map = {}

        if s_length != t_length:
            final_arr.append('NO')
            break

        for j in range(s_length):
            s_element = s_current_array[j]
            t_element = t_current_array[j]

            s_map[s_element] = s_map.get(s_element, 0) + 1
            t_map[t_element] = t_map.get(t_element, 0) + 1

        for element, value in s_map.items():
            t_value = t_map.get(element, 0)
            difference = abs(value - t_value)

            if difference <= 3:
                continue
            else:
                final_arr.append('NO')
                break
        else:
            final_arr.append('YES')

        s_map.clear()
        t_map.clear()

    return final_arr